mso/resource_mso_fabric_policies_vlan_pool.go

```diff
@@ -0,0 +1,266 @@
+package mso
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/ciscoecosystem/mso-go-client/client"
+	"github.com/ciscoecosystem/mso-go-client/container"
+	"github.com/ciscoecosystem/mso-go-client/models"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func resourceMSOVlanPool() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceMSOVlanPoolCreate,
+		Read:   resourceMSOVlanPoolRead,
+		Update: resourceMSOVlanPoolUpdate,
+		Delete: resourceMSOVlanPoolDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceMSOVlanPoolImport,
+		},
+
+		SchemaVersion: 1,
+		Schema: map[string]*schema.Schema{
+			"template_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 64),
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"uuid": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"allocation_mode": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"dynamic", "static",
+				}, false),
+			},
+			"vlan_range": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"from": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"to": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"allocation_mode": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Computed: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"dynamic", "static",
+							}, false),
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func setVlanRange(rangeEntries *schema.Set) []map[string]any {
+	rangeEntryList := rangeEntries.List()
+	VlanRange := make([]map[string]any, len(rangeEntryList))
```

Comment:
is there a reason for the capitalised variable name?
```suggestion
	vlanRange := make([]map[string]any, len(rangeEntryList))
```